### PR TITLE
Adds social media share icon button component

### PR DESCRIPTION
**Pull request by phuson**
Supports Facebook, Twitter, LinkedIn, and Google Plus social media sharing.

### CODE REVIEW

File: src/js/components/SocialShare.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+SocialShare.propTypes = {
+  type: PropTypes.oneOf(['facebook', 'twitter', 'linkedin', 'google']),
+  title: PropTypes.string,
+  text: PropTypes.string,
+  link: PropTypes.string.isRequired
+};
+
+SocialShare.defaultProps = {
+  type: 'facebook'
```
Comment (ericsoderberghp) [inline src/js/components/SocialShare.js]: I'm thinking we shouldn't default the type and require callers to specify which type they want.

File: src/js/components/SocialShare.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    return (
+      <Anchor href={href} icon={socialIcon} target="_blank" />
+    );
+  }
+};
+
+SocialShare.propTypes = {
+  type: PropTypes.oneOf(['facebook', 'twitter', 'linkedin', 'google']),
```
Comment (ericsoderberghp) [inline src/js/components/SocialShare.js]: The type should be required.

**Comment by ericsoderberghp** [comment]
Just two minor comments.

**Comment by phuson** [comment]
Okay, made `type` prop a required prop. Updated grommet docs to reflect this. Thanks!

**Pull request merged**